### DISCUSSION TITLE
Is global-variable-not-assigned (PLW0602) auto fixable?

### DISCUSSION THREAD

**Discussion by harupy** [astral-sh/ruff#14031] [Q&A] [answered]
Is https://docs.astral.sh/ruff/rules/global-variable-not-assigned/#global-variable-not-assigned-plw0602 auto fixable?

**Comment by MichaReiser** [comment, marked as answer]
No, it's not. The reason is that there's no obvious fix. The fix is either:

* To remove the globals 
* To assign to it 

but Ruff doesn't know which one it is. The documentation mentions when a fix is available. E.g. [`iteration-over-set`](https://docs.astral.sh/ruff/rules/iteration-over-set/) states that it always has a fix.

**Reply by harupy** [reply]
Makes sense, thanks for the clarification!
